Documentation/design/initphase.md

```diff
@@ -0,0 +1,44 @@
+# Init Phase
+
+## Goals
+
+1. Generates `InstallConfig` by prompting user for required parameters, and populating the rest of configuration with default values.
+2. Make useful decisions on default values for most options.
+3. Init phase should never prompt the user for more than 10 options unless there is a very strong reason.
+
+## Non Goals
+
+1. Supporting all the configuration option through init is not a goal. Init is just a helper for humans.
+
+## Overview
+
+The output of the `init` command is a config file that describes a default cluster for the given provider, and is valid/immediately usable.
+
+The `init` step is not required, this is merely a helper to generate a valid `InstallConfig`. A user could use examples from documentation, roll their own by hand, or re-use existing configs from source control.
+
+## Detailed Design
+
+### Prompts on CLI vs flags vs Environment variables
+
+1. Flags don’t work well here because we want the fast-path to be single command.
```

Comment:
Is this clear to other's as a stand-alone statement? I had this in the design doc - but I we may need to pair with examples (e.g. if it's flag based, then the fast path of `install-openshift launch` would have to plumb through all `init` flags: `install-openshift launch --base-domain=example.com` (also some options are provider specific which makes flag handling tedious).

Reply:
i'll add some examples.